addons/probation_bot.py

```diff
@@ -0,0 +1,25 @@
+from discord.ext import commands
+import hashlib
+
+class ProbationBot:
+    """
+    Kicking and banning users.
+    """
+    def __init__(self, bot):
+        self.bot = bot
+        print('Addon "{}" loaded'.format(self.__class__.__name__))
+
+    async def on_message(self, message):
+        try:
+            if message.channel == self.bot.welcome_channel:
+                member = message.author
+                full_name = member.name + '#' + member.discriminator
```

Comment:
You might want to use str(member) here instead of the member.name member.discriminator, it effectively returns the same. (At least it does in rewrite, I have never tested this on latest).

Reply:
I'd rather this stays explicit, as to make it clear what the exact format of the sha1 needs to look like :).